build/azure-pipelines/darwin/sql-product-build-darwin-universal.yml

```diff
@@ -0,0 +1,172 @@
+
+  - script: |
+      set -e
+      git remote add distro "https://github.com/$(VSCODE_MIXIN_REPO).git"
+      git fetch distro
+      git merge $(node -p "require('./package.json').distro")
+    displayName: Merge distro
+
+  - script: |
+      mkdir -p .build
+      node build/azure-pipelines/common/sql-computeNodeModulesCacheKey.js x64 > .build/yarnlockhash
+    displayName: Prepare yarn cache key
+
+  - task: Cache@2
+    displayName: Restore Cache - Node Modules
+    inputs:
+      key: 'nodeModules | $(Agent.OS) | .build/yarnlockhash'
+      path: .build/node_modules_cache
+      cacheHitVar: NODE_MODULES_RESTORED
+    continueOnError: true
+
+  - script: |
+      set -e
+      tar -xzf .build/node_modules_cache/cache.tgz
+    condition: and(succeeded(), eq(variables.NODE_MODULES_RESTORED, 'true'))
+    displayName: Extract node_modules archive
+
+  - script: |
+      set -e
+      CHILD_CONCURRENCY=1 yarn --frozen-lockfile
+    displayName: Install dependencies
+    env:
+      GITHUB_TOKEN: $(github-distro-mixin-password)
+    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
+
+  - script: |
+      set -e
+      node build/azure-pipelines/common/listNodeModules.js .build/node_modules_list.txt
+      mkdir -p .build/node_modules_cache
+      tar -czf .build/node_modules_cache/cache.tgz --files-from .build/node_modules_list.txt
+    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
+    displayName: Create node_modules archive
+
+  - script: |
+      set -e
+      yarn postinstall
+    displayName: Run postinstall scripts
+    condition: and(succeeded(), eq(variables.NODE_MODULES_RESTORED, 'true'))
+
+  - script: |
+      set -e
+      node build/azure-pipelines/mixin
+    displayName: Mix in quality
+
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download arm64 and x64 packages'
+    inputs:
+      downloadType: specific
+      itemPattern: 'drop/darwin/archive/azuredatastudio-darwin-@(arm64|x64)-unsigned.zip'
+      downloadPath: $(agent.builddirectory)
+
+  - script: |
+      set -e
+      unzip $(agent.builddirectory)/drop/darwin/archive/azuredatastudio-darwin-x64-unsigned.zip -d $(agent.builddirectory)/azuredatastudio-darwin-x64
+      unzip $(agent.builddirectory)/drop/darwin/archive/azuredatastudio-darwin-arm64-unsigned.zip -d $(agent.builddirectory)/azuredatastudio-darwin-arm64
+      DEBUG=* node build/darwin/create-universal-app.js
+    displayName: Create Universal App
+
+  - script: |
+      set -e
+      pushd $(agent.builddirectory)/azuredatastudio-darwin-$(VSCODE_ARCH)
+      ls
+
+      echo "Cleaning the application"
+      xattr -cr *.app
+      cd *.app
+      find . -name '._*' -print0 | xargs -0 rm -rf --
+      cd ..
+
+      echo "Signing the application with deep"
+      codesign --deep --force --timestamp --options runtime --entitlements $(Build.SourcesDirectory)/build/azure-pipelines/darwin/entitlements.xml -s LPV3BJJYXS *.app
```

Comment:
What's this LPV3BJJYXS value from?

Reply:
this piece of code is copied from the sql-product-build-darwin.yml, I can try to consolidate them into one file.

Reply:
Might not be a bad idea. And if you could find out what this value is and document that in the code that'd be great.

Reply:
1. merged the universal yml with the main one - queueing a new build to verify the changes.
2. created an issue to track the signing tasks cleaning/documenting work: https://github.com/microsoft/azuredatastudio/issues/20797, Karl forwarded me something that might help.